webapp/app/templates/lotse/form_steuernummer.html

```diff
@@ -0,0 +1,184 @@
+                    deleteInput(requestNewTaxNumberCheckbox);
+                    hideRequiredInput(requestNewTaxNumberCheckbox);
+                    break;
+                case TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists:
+                    showRequiredInput(stateDropdown);
+                    deleteInput(bufaNrDropdown);
+                    hideRequiredInput(bufaNrDropdown);
+                    deleteInput(taxNumberInput);
+                    hideRequiredInput(taxNumberInput);
+                    deleteInput(requestNewTaxNumberCheckbox);
+                    hideRequiredInput(requestNewTaxNumberCheckbox);
+                    break;
+                case TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists:
+                    showRequiredInput(stateDropdown);
+                    deleteInput(bufaNrDropdown);
+                    hideRequiredInput(bufaNrDropdown);
+                    deleteInput(taxNumberInput);
+                    hideRequiredInput(taxNumberInput);
+                    deleteInput(requestNewTaxNumberCheckbox);
+                    hideRequiredInput(requestNewTaxNumberCheckbox);
+                    break;
+                case TAX_NUMBER_FORM_STATES.TaxOfficeSelection:
+                    showRequiredInput(stateDropdown);
+                    deleteInput(bufaNrDropdown);
+                    showRequiredInput(bufaNrDropdown);
+                    deleteInput(taxNumberInput);
+                    hideRequiredInput(taxNumberInput);
+                    deleteInput(requestNewTaxNumberCheckbox);
+                    hideRequiredInput(requestNewTaxNumberCheckbox);
+                    break;
+                case TAX_NUMBER_FORM_STATES.TaxNumberInput:
+                    showRequiredInput(stateDropdown);
+                    deleteInput(bufaNrDropdown);
+                    hideRequiredInput(bufaNrDropdown);
+                    showRequiredInput(taxNumberInput);
+                    deleteInput(requestNewTaxNumberCheckbox);
+                    hideRequiredInput(requestNewTaxNumberCheckbox);
+                    break;
+                case TAX_NUMBER_FORM_STATES.RequestNewTaxNumber:
+                    showRequiredInput(stateDropdown);
+                    showRequiredInput(bufaNrDropdown);
+                    deleteInput(taxNumberInput);
+                    hideRequiredInput(taxNumberInput);
+                    showRequiredInput(requestNewTaxNumberCheckbox);
+                    break;
+            }
+        }
+
+        function runThroughAllStates(){
+            changeTaxNumberExists();
+            changeBundesland();
+            changeTaxOffice();
+        }
+        document.addEventListener('DOMContentLoaded', () => {
+            // Register event handlers
+            document.querySelectorAll('input[type=radio][name=steuernummer_exists]').forEach((field)=>{
+                field.addEventListener("click",() => {changeTaxNumberExists(); runThroughAllStates(); handleState();});
```

Comment:
Why do you use `click` here instead of `change`? I think we could get rid of the `wasCheckedBefore` attribute if we change this?

Reply:
See the answer below